db/models/applications.js

```diff
@@ -0,0 +1,238 @@
+// @ts-nocheck
+'use strict';
+const moment = require('moment');
+const { Model, Op } = require('sequelize');
+
+class PendingApplication extends Model {
+    static init(sequelize, DataTypes) {
+        const schema = {
+            id: {
+                primaryKey: true,
+                type: DataTypes.UUID,
+                defaultValue: DataTypes.UUIDV4,
+                allowNull: false
+            },
+
+            /**
+             * User model reference
+             */
+            userId: {
+                type: DataTypes.INTEGER,
+                references: { model: 'users', key: 'id' }
+            },
+
+            /**
+             * Form model ID
+             * e.g. awards-for-all
+             */
+            formId: {
+                type: DataTypes.STRING,
+                allowNull: false
+            },
+
+            /**
+             * Temporary JSON snapshot of form data
+             * Raw output from joi schema
+             */
+            applicationData: {
+                type: DataTypes.JSON,
+                allowNull: true
+            },
+
+            /**
+             * Track submission attempts
+             * If there is an error in submission this will be incremented,
+             * allowing us to report on failed submission attempts.
+             */
+            submissionAttempts: {
+                type: DataTypes.INTEGER,
+                defaultValue: 0,
+                allowNull: false
+            },
+
+            /**
+             * Expiry date for pending application
+             * A scheduled job runs which cleans out expired applications
+             */
+            expiresAt: {
+                type: DataTypes.DATE,
+                allowNull: false
+            }
+        };
+
+        return super.init(schema, {
+            sequelize
+        });
+    }
+    static findAllByForm({ userId, formId }) {
+        return this.findAll({
+            where: {
+                userId: { [Op.eq]: userId },
+                formId: { [Op.eq]: formId }
+            },
+            order: [['updatedAt', 'DESC']]
+        });
+    }
+    static findApplicationForForm({ formId, applicationId, userId }) {
+        return this.findOne({
+            where: {
+                id: { [Op.eq]: applicationId },
+                formId: { [Op.eq]: formId },
+                userId: { [Op.eq]: userId }
+            }
+        });
+    }
+    static createNewApplication({ userId, formId }) {
+        // @TODO: Should this be defined in config?
+        const expiresAt = moment()
+            .add('3', 'months')
+            .toDate();
+
+        return this.create({
+            userId: userId,
+            formId: formId,
+            applicationData: null,
+            expiresAt: expiresAt
+        });
+    }
+    static saveApplicationState(id, data) {
```

Comment:
Should this method effectively reset the `expiresAt` date each time it's called?